include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -1502,6 +1502,14 @@ struct OrtCustomOp {
   OrtCustomOpInputOutputCharacteristic(ORT_API_CALL* GetOutputCharacteristic)(_In_ const struct OrtCustomOp* op, _In_ size_t index);
 };
 
+/*
+ * This is the old way to add the CUDA provider to the session, please use SessionOptionsAppendExecutionProvider_CUDA above to access the latest functionality
+ * This function always exists, but will only succeed if Onnxruntime was built with CUDA support and the CUDA provider shared library exists
+ * 
+ * \param device_id cuda device id, starts from zero.
+*/
+ORT_API_STATUS(OrtSessionOptionsAppendExecutionProvider_CUDA, _In_ OrtSessionOptions* options, int device_id);
```

Comment:
It is confusing.  We are adding this function to all EPs.
First, do we need to keep this? For what, backward compatibility? If it is important to the CUDA EP, then why it is not important to the other EPs? By doing this, you are breaking it for the others EPs.

Reply:
How does this break other EPs?

Reply:
Ah, I was wrong. We are using def files, so this function won't be auto exported.